src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -377,8 +401,6 @@ private bool TryClose()
 
                 await Input.CompleteAsync();
 
-                await RequestBodyPipe.Writer.CompleteAsync();
```

Comment:
Removed because errors weren't being correctly reported by request pipe writer. `CompleteAsync` was being called here before `CompleteAsync(Exception)` was being called somewhere else later.

Reply:
Where is the call to CompleteAsync(Exception)? I imagine it's due to abort being called somehow?

Reply:
`RequestBodyPipe.Writer.Complete(abortReason);` is called in the new `AbortCore`.

Reply:
Is AbortCore called on each stream now?

Reply:
Do you mean from here?

https://github.com/dotnet/aspnetcore/blob/bb0be8c3148d2b28778eb313da6643cf4000f4ea/src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs#L353-L356

Yes

Reply:
Hm, maybe that shouldn't be happening...

Reply:
I don't know enough about this area to offer a good opinion, but I looked at `Http2Connection` and it calls into equivilent functionality on `Http2Stream`.

Reply:
The biggest difference is Abort in HTTP/2 keeps track of whether it has already aborted - https://github.com/dotnet/aspnetcore/blob/bb0be8c3148d2b28778eb313da6643cf4000f4ea/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs#L532-L538

I left a TODO comment in Http3Stream.Abort about whether this needs to be tracked.